server/controllers/article_controller.js

```diff
@@ -83,9 +83,9 @@ export const getArticleAnnotations = (user, uri, topLevelOnly) => {
       return article.annotations;
     });
   } else {
-    const deepPath = 'annotations.childAnnotations.childAnnotations.childAnnotations.childAnnotations.childAnnotations.childAnnotations';
+    const deepPath = 'annotations'.concat('.childAnnotations'.repeat(50));
     return getArticle(uri)
-    .deepPopulate(deepPath, { populate: { annotations: { match: query } } })
+    .deepPopulate(deepPath, { match: query })
```

Comment:
I think it's because it's not doing the `match: { parent: null }`, which you had